examples/clouds-configmap.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: clouds-configmap
+  namespace: openstack
+data:
+  clouds.yaml: |
+    clouds:
+      devstack-admin:
+        auth:
+          auth_url: http://192.168.122.4/identity
+          password: password
```

Comment:
I agree w/ John here, I would rather document in the `README` that we can build a secret like the following:

```
---
apiVersion: v1
kind: Secret
metadata:
  name: clouds-secret
  namespace: openstack
stringData:
  clouds.yaml: |
     clouds:
      devstack-admin:
        auth:
          auth_url: http://192.168.122.4/identity
          password: password
          ...
          ...

```

As long as the ctlplane is up, the `clouds.yaml` file can be retrieved with the following:

` oc get cm openstack-config -o json | jq -r '.data["clouds.yaml"]' `

and used to build the secret above. I would invest some time to properly document what we're doing and help people onboarding to the operator (especially for a dev workflow).

@cloudguruab does this make sense?

Reply:
Yes, makes a ton of sense I think my initial approach here was to just get something on paper and then refine the work. I am not too familiar with secrets in k8s so I'll read up on some documentation. If we are creating a configmap using the operator wouldnt this information need to sit inside a folder somewhere? What would the workflow look like for the end user if this were to be a secret and they needed to fetch this information to use in the pod? @fmount

Reply:
@cloudguruab Here's an example of making an SSH key secret:

https://github.com/fultonj/zed/blob/main/crc/cr/ssh_secret.sh

Here's a commit I did to modify my ansible play to use the ssh secret in place of a configmap mount:

https://github.com/fultonj/zed/commit/3c0b354231e71a5eb8cd92df2d2600a01a4f8d34#diff-0aed6d775e7d91241f8d8db3ed85063ab66a9e7d853bfbd218331893b3c4bce4

So in the above I didn't mount a configmap with my SSH key to get the key into my environment as the following was removed and replaced with the extraMounts you can see.

    - name: key-configmap
      mountpath: /runner/env

Reply:
> If we are creating a configmap using the operator wouldnt this information need to sit inside a folder somewhere? 

Yes, in this patch there's an example configmap. You could replace it with a example secret. It's very similar to a config map except the contents are hidden. 

> What would the workflow look like for the end user if this were to be a secret and they needed to fetch this information to use in the pod?

So instead of creating the configmap in advance with `oc apply -f configmap.yml` they'd create it with `oc apply -f secret.yml` and provided they did that, the next CR could reference it.